### ISSUE TITLE
DOI's not activating

### ISSUE THREAD

**Issue by martinseul** [hydroshare/hydroshare#4261]
There seems t be an issue with the publishing process. The process does not complete and the status is stuck on pending we had 2  user point out that their resources are still waiting for a DOI. 

https://www.hydroshare.org/resource/6d6216f4973c45b6be80b3ce5e3e6764/
and 
(https://www.hydroshare.org/resource/a2f0d5fd10f14cd189a3465f72cba6f3/) 

We also received a message from crossref yesterday pointing out that several resources are waiting "activation"

Published resource DOI https://doi.org/10.4211/hs.https://doi.org/10.4211/hs.cb522bbb0f6449ac9d6ca2c189e5c604 is not yet activated with request data deposited since 12/14/2020.
Published resource DOI https://doi.org/10.4211/hs.ade683be61e54fa5b60da97418a5f3df is not yet activated with request data deposited since 06/01/2021.
Published resource DOI https://doi.org/10.4211/hs.0cdad4c81d2545408e43867d48fedfe9 is not yet activated with request data deposited since 05/19/2021.
Published resource DOI https://doi.org/10.4211/hs.6d479a668f6c40a289d09e6c95ecf250 is not yet activated with request data deposited since 06/03/2021.
Published resource DOI https://doi.org/10.4211/hs.31b7c356119241e0a9b1f6f56a9cb44e is not yet activated with request data deposited since 06/03/2021.
Published resource DOI https://doi.org/10.4211/hs.86225d08252747d5a78477c1f74eb158 is not yet activated with request data deposited since 05/21/2021.
Published resource DOI https://doi.org/10.4211/hs.bb01ce1e21214f93b197d6855823af53 is not yet activated with request data deposited since 06/02/2021.
Published resource DOI https://doi.org/10.4211/hs.92999529572340c593b2b8c5b5bcd211 is not yet activated with request data deposited since 05/28/2021.
Published resource DOI https://doi.org/10.4211/hs.d001620c2f104a0891dc18825ea2fe31 is not yet activated with request data deposited since 06/03/2021.
Published resource DOI https://doi.org/10.4211/hs.6d6216f4973c45b6be80b3ce5e3e6764 is not yet activated with request data deposited since 06/07/2021.

I noticed that this goes back to 5/19 a day after we released 1.44, Looks like we have a regression.

**Comment by hyi** [comment]
@martinseul @dtarb @sblack-usu I have fixed all the pending publication resources noted above. One resource as noted in ```Published resource DOI https://doi.org/10.4211/hs.https://doi.org/10.4211/hs.cb522bbb0f6449ac9d6ca2c189e5c604 is not yet activated with request data deposited since 12/14/2020.``` is caused by a bug which only occurs when the first time crossref metadata deposition failed, and this bug has been fixed. The pending publications for all the other resources noted above are regressions introduced by the PR for allowing editing metadata for published resources. The fix for the regression has been included in the PR as well here https://github.com/hydroshare/hydroshare/pull/4262 and I have assigned Scott to code review. In fact, this regression revealed a bigger issue across HydroShare codebase. Specifically, ```__debug__``` global variable is being used to check whether it is production or dev server mainly to print a lot of debug messages into log, however, in our production environment, this global variable is True as detailed in this post: https://stackoverflow.com/questions/5029316/what-sets-the-debug-variable-in-a-production-django-wsgi-based-site. To really differentiate between production and dev environment, ```settings.DEBUG``` variable should be used instead of ```__debug__``` global variable. The regression for pending publication was caused by using this global debug variable and only calling the method to deposit publication resource's metadata to crossref when this ```__debug__``` variable is False wrongly assuming this variable is False only in our production server. I have manually fixed all those resources on production server to deposit their metadata to crossref and verified they have all been deposited correctly. After my PR is merged, this regression issue should be fixed.
